runtime/client_http_request.go

```diff
@@ -203,21 +212,80 @@ func (req *ClientHTTPRequest) Do() (*ClientHTTPResponse, error) {
 		/* coverage ignore next line */
 		return nil, err
 	}
+	var retryCount int64 = 1
+	var res *http.Response
+
+	//when timeoutAndRetryOptions per request is not configured, use default client level timeout
+	if req.timeoutAndRetryOptions == nil || req.timeoutAndRetryOptions.MaxAttempts == 0 {
+		res, err = req.client.Client.Do(req.httpReq.WithContext(ctx))
+	} else {
+		res, retryCount, err = req.executeDoWithRetry(ctx) //new code for retry and timeout per ep level
+	}
 
-	res, err := req.client.Client.Do(req.httpReq.WithContext(ctx))
 	span.Finish()
+
 	if err != nil {
-		req.ContextLogger.ErrorZ(req.ctx, "Could not make outbound request", zap.Error(err))
-		return nil, err
+		req.ContextLogger.ErrorZ(req.ctx, fmt.Sprintf("Could not make http outbound %s.%s request",
```

Comment:
is the formatting in the error log message necessary? Shouldn't we get this information through some of the logger tags from the context logger?

Reply:
Ideally, it should have come through the context logger but it wasn't the case. As a quick fix I added it here instead debugging it